nvs_partition_gen.py

```diff
@@ -0,0 +1,892 @@
+            data_chunk = data[offset:offset + chunk_size]
+
+            # Compute CRC of data chunk
+            struct.pack_into('<H', entry_struct, 24, chunk_size)
+
+            if type(data) != bytes:
+                data_chunk = bytes(data_chunk, encoding='utf8')
+
+            crc = zlib.crc32(data_chunk, 0xFFFFFFFF)
+            struct.pack_into('<I', entry_struct, 28, crc & 0xFFFFFFFF)
+
+            # compute crc of entry header
+            entry_struct = self.set_crc_header(entry_struct)
+
+            # write entry header
+            self.write_entry_to_buf(entry_struct, 1,nvs_obj)
+            # write actual data
+            self.write_entry_to_buf(data_chunk, datachunk_entry_count,nvs_obj)
+
+            chunk_count = chunk_count + 1
+
+            if remaining_size or (tailroom - chunk_size) < Page.SINGLE_ENTRY_SIZE:
+                nvs_obj.create_new_page()
+                self = nvs_obj.cur_page
+
+            offset = offset + chunk_size
+
+            # All chunks are stored, now store the index
+            if not remaining_size:
+                # Initialise data field to 0xff
+                data_array = bytearray(b'\xff') * 8
+                entry_struct[24:32] = data_array
+
+                # change type of data to BLOB_IDX
+                entry_struct[1] = Page.BLOB_IDX
+
+                # Set Span
+                entry_struct[2] = 1
+
+                # Update the chunkIndex
+                chunk_index = Page.CHUNK_ANY
+                entry_struct[3] = chunk_index
+
+                struct.pack_into('<I', entry_struct, 24, data_size)
+                entry_struct[28] = chunk_count
+                entry_struct[29] = chunk_start
+
+                # compute crc of entry header
+                entry_struct = self.set_crc_header(entry_struct)
+
+                # write last entry
+                self.write_entry_to_buf(entry_struct, 1,nvs_obj)
+                break
+
+        return entry_struct
+
+    def write_single_page_entry(self, entry_struct, data, datalen, data_entry_count, nvs_obj):
+        # compute CRC of data
+        struct.pack_into('<H', entry_struct, 24, datalen)
+
+        if type(data) != bytes:
+            data = bytes(data, encoding='utf8')
+
+        crc = zlib.crc32(data, 0xFFFFFFFF)
+        struct.pack_into('<I', entry_struct, 28, crc & 0xFFFFFFFF)
+
+        # compute crc of entry header
+        entry_struct = self.set_crc_header(entry_struct)
+
+        # write entry header
+        self.write_entry_to_buf(entry_struct, 1, nvs_obj)
+        # write actual data
+        self.write_entry_to_buf(data, data_entry_count, nvs_obj)
+
+    """
+    Low-level function to write variable length data into page buffer. Data should be formatted
+    according to encoding specified.
+    """
+    def write_varlen_data(self, key, data, encoding, ns_index,nvs_obj):
+        # Set size of data
+        datalen = len(data)
+
+        if datalen > Page.PAGE_PARAMS["max_new_blob_size"]:
```

Comment:
This line used to read `if datalen > Page.PAGE_PARAMS["max_old_blob_size"]:` which was the previous 1984 byte limit which isn't correct for the newer 3.3.2 esp-idf we are using.